README.md

```diff
@@ -1,27 +1,395 @@
+Persistent Volumes. We propose two new Custom Resources to abstract an object store bucket
+and a claim/request for such a bucket.  It's important to keep in mind that this proposal 
+only defines bucket and bucket claim APIs and related library code. The lib ensures that 
+the _contract_ made to app developers regarding the artifacts of bucket creation (Secrets,
+ConfigMaps) are guaranteed. The actual creation of buckets belongs to each object store
+provisioner. However, as is true for dynamic Persistent Volume provisioning, each bucket
+provisioner needs to adhere to the inferfaces defined in the bucket library.
 
-Then install the dependencies
+An `ObjectBucketClaim` (OBC) is similar in usage to a Persistent Volume Claim
+and an `ObjectBucket` (OB) is the Persistent Volume equivalent. 
+Bucket binding refers to the actual bucket being created by the underlying object
+store provide, and the generation of artifacts which will be consumed by application pods.
+An OBC is namespaced and references a storage class which defines the object store. The
+details of the object store (ceph, minio, cloud, on-prem) are not visible to the app pod
+and can change without disturbing the app in anyway. An OB is non-namespaced (global),
+typically not visible to end users, and will contain info pertinent to the provisioned
+bucket. Like PVs, there is a 1:1 binding of an OBC to an OB.
 
-`dep ensure -v`
+As is true for dynamic PV provisioning, a bucket provisioner needs to be running
+for each object store supported by the Kubernetes cluster. For example, if the
+underlying object store is AWS S3, the developer will create an OBC, referencing
+a Storage Class which references the S3 store. The cluster has the S3 provisioner
+running which is watching for OBCs that it knows how to handle. Other OBCs are ignored
+by the S3 provisioner. Additionally, the same cluster can have a rook-ceph RGW
+provisioner running which also watches OBCs. Like the S3 proivisioner, it only handles
+OBCs that it knows how to provision and skips the rest. In this proposal, the bucket
+provisioners will be simple-to-write operators because the bucket provisioning lib
+handles the bulk of the work. Each provisioner is only responsible for writing
+`Provision()` and `Delete()`functions (the _business logic_) and a short `main()`
+function. **Note:** even though the PV-PVC design supports static provisioning, only
+dynamic provisioning is supported by the bucket lib.
 
-## Update generated code
+The `Provision()` and `Delete()`functions are interfaces defined in the bucket library.
+All bucket provisioners are required to return an OB struct and the bucket
+credentials (access-key, endpoint) as defined in the bucket lib. Based on the returned
+OB, the lib will create the bucket artifacts: user Secret, user ConfigMap, and OB
+controller (primarily for status updates). The Secret and ConfigMap have deterministic
```

Comment:
>  and OB controller (primarily for status updates)

This reads as though the OB controller(s?) is created in response to an OBC create.